### PR TITLE
Task 5.2

### PR DESCRIPTION
**Pull request by fujoshik**

### CODE REVIEW

File: Task 5.2/Task5.2/Entities/Book.cs [hunk 1]
```diff
@@ -0,0 +1,26 @@
+﻿namespace Task5._2.Entities
+{
+    public class Book
+    {
+        public string Title { get; set; }
+        public DateTime? PublicationDate { get; set; }
```
Comment (IgorKheidorov) [inline Task 5.2/Task5.2/Entities/Book.cs:6]: The set of authors of the book (a collection of non-repeating strings, possibly empty).-> to use list is not a good idea

File: Task 5.2/Task5.2/Entities/Catalog.cs [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+            _books = new Dictionary<string, Book>();
+        }
+
+        public void AddBook(string isbn, Book book)
+        {
+            if (book == null)
+            {
+                throw new ArgumentNullException(nameof(book));
+            }
+
```
Comment (IgorKheidorov) [inline Task 5.2/Task5.2/Entities/Catalog.cs:21]: Not proper logics,  acceptable format is  XXX-X-XX-XXXXXX-X, but not ----XXXXXXXXXXXXX, but you don't take this into considereation

File: Task 5.2/Task5.2/Entities/Catalog.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            if (!((isbn.Count(x => x == '-') == 4 && isbn.Length == 17) || (isbn.Length == 13 && !isbn.Contains("-"))))
+            {
+                throw new ArgumentException(nameof(isbn));             
+            }
+
+            string cleanIsbn = isbn.Replace("-", "");
+            _books.Add(cleanIsbn, book);
+        }
+
```
Comment (IgorKheidorov) [inline Task 5.2/Task5.2/Entities/Catalog.cs:30]: check input string for isbn validity

**Review by IgorKheidorov — CHANGES_REQUESTED** [review]
Please fix comments

File: Task 5.2/Task5.2/Entities/Catalog.cs (hunk 3 above)
Comment (fujoshik) [reply Task 5.2/Task5.2/Entities/Catalog.cs:30]: Fixed

File: Task 5.2/Task5.2/Entities/Catalog.cs (hunk 2 above)
Comment (fujoshik) [reply Task 5.2/Task5.2/Entities/Catalog.cs:21]: Fixed

File: Task 5.2/Task5.2/Entities/Book.cs (hunk 1 above)
Comment (fujoshik) [reply Task 5.2/Task5.2/Entities/Book.cs:6]: Fixed

File: Task 5.2/Task5.2/Entities/Catalog.cs [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+            {
+                throw new ArgumentException(nameof(isbn));
+            }
+
+            string cleanIsbn = isbn.Replace("-", "");
+            _books.Add(cleanIsbn, book);
+        }
+
+        private bool IsInCorrectFormat(string isbn)
+        {
```
Comment (IgorKheidorov) [inline Task 5.2/Task5.2/Entities/Catalog.cs:32]: Try to create Regex to check isbn, too complicated logcs

**Review by IgorKheidorov — APPROVED** [review]
Please try to create Regex for isbn check and can merge

**Pull request merged**